### PR TITLE
Don't index resource template resource templates.

### PR DESCRIPTION
**Pull request by justinlittman**
refs https://github.com/LD4P/sinopia_editor/issues/2378

### CODE REVIEW

**Comment by justinlittman** [comment]
Tested locally (In addition to unit test.)

File: __tests__/TemplateIndexer.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { datasetFromJsonld } from 'utilities'
+import TemplateIndexer from 'TemplateIndexer'
+
+describe('Indexer', () => {
+  describe('Indexing a resource template resource template', () => {
```
Comment (jermnelson) [inline __tests__/TemplateIndexer.test.js:5]: I find the construction `resource template resource template` to be confusing. Could this be adjusted as `resource template's resource template` or maybe `base resource template`?

File: src/TemplateIndexer.js [hunk 2]
```diff
@@ -18,7 +18,10 @@ export default class {
    * @returns {Object} an object containing configured field values if any found
    */
   index() {
-    this.indexObject.id = this.valueFor('http://sinopia.io/vocabulary/hasResourceId')
+    const resourceId = this.valueFor('http://sinopia.io/vocabulary/hasResourceId')
+    // Don't index resource template resource templates.
```
Comment (jermnelson) [inline src/TemplateIndexer.js:22]: See comment above about `resource template resource template`

File: __tests__/TemplateIndexer.test.js (hunk 1 above)
Comment (justinlittman) [reply __tests__/TemplateIndexer.test.js]: Clarified.

File: src/TemplateIndexer.js (hunk 2 above)
Comment (justinlittman) [reply src/TemplateIndexer.js]: Clarified.

**Review by jermnelson — APPROVED** [review]

**Pull request merged**